docs/DeveloperGuide.adoc

```diff
@@ -279,75 +279,185 @@ Refer to the guide <<DevOps#, here>>.
+|        5 |          5 | User who likes sharing my cooking                 | post my recipes and dishes on social media                                       | share recipes and images for others to use
+|        5 |          5 | User who appreciates efficiency                   | add a recipe directly from online                                                | I can be efficient
+|===
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is `CookBuddy` and the *Actor* is the `User`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: List recipes
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  `User` requests to list recipes
+2.  `CookBuddy` displays the list of recipes
 +
 Use case ends.
 
 *Extensions*
 
+[none]
+* 1a. The name cannot be found, or the index is invalid.
++
+[none]
+** 1a1. `CookBuddy` throws an error message.
++
+Use case resumes at step 1.
+
 [none]
 * 2a. The list is empty.
 +
+[none]
+** 2a1. `CookBuddy` displays a message stating the list is empty
++
+Use case ends.
+
+
+
+[discrete]
+=== Use case: Delete recipe
+
+*MSS*
+
+1.  `User` requests to delete a specific _recipe_ by specifying its name or index
+2.  CookBuddy deletes the recipe
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The name cannot be found, or the index is invalid.
++
+[none]
+** 1a1. `CookBuddy` throws an error message.
++
+Use case ends.
+
+
+[discrete]
+=== Use case: Modify recipe
+
+*MSS*
+
+1. `User` requests to modify a recipe
+2. `CookBuddy` edits attributes of the recipe, and asks for user confirmation
+3. `User` confirms the edit
++
 Use case ends.
 
-* 3a. The given index is invalid.
+*Extensions*
+
+[none]
+* 1a. `User` does not provide new attributes.
+[none]
+** `CookBuddy` throws an error message.
++
+Use case resumes at step 1.
+* 2a. `User` does not confirm.
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 2a1. `CookBuddy` does not save the edit
 +
-Use case resumes at step 2.
+Use case ends.
+
 
-_{More to be added}_
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be able to hold up to 1000 recipes without noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```

Comment:
You might want to state an NFR specific to your app instead of addressbook.
All NFRs are too generic.